src/lib/components/layout/QLayout.svelte

```diff
@@ -22,116 +20,131 @@
     fixed: boolean;
   }
 
-  export type LayoutContext = Readable<{
+  export type LayoutContext = {
     header?: AppbarContext;
     footer?: AppbarContext;
     drawerLeft: DrawerContext;
     drawerRight: DrawerContext;
-  }>;
+  };
 </script>
 
 <script lang="ts">
-  export let view: QLayoutProps["view"] = "hhh lpr fff",
-    headerHeight: QLayoutProps["headerHeight"] = "64px",
-    footerHeight: QLayoutProps["footerHeight"] = "80px",
-    leftDrawerWidth: QLayoutProps["leftDrawerWidth"] = "300px",
-    rightDrawerWidth: QLayoutProps["rightDrawerWidth"] = "300px",
-    leftRailbarWidth: QLayoutProps["leftRailbarWidth"] = "88px",
-    rightRailbarWidth: QLayoutProps["rightDrawerWidth"] = "88px",
-    userClasses: QLayoutProps["userClasses"] = undefined,
-    userStyles: QLayoutProps["userStyles"] = undefined;
-  export { userClasses as class, userStyles as style };
+  let {
+    view = "hhh lpr fff",
+    headerHeight = "64px",
+    footerHeight = "80px",
+    leftDrawerWidth = "300px",
+    rightDrawerWidth = "300px",
+    leftRailbarWidth = "88px",
+    rightRailbarWidth = "88px",
+    content,
+    railbarLeft,
+    railbarRight,
+    drawerLeft,
+    drawerRight,
+    header,
+    footer,
+    onscroll,
+    onresize,
+    ...props
+  }: QLayoutProps = $props();
 
-  $: style = createStyles(
-    {
-      "--header-height": isNaN(Number(headerHeight)) ? headerHeight : `${headerHeight}px`,
-      "--footer-height": isNaN(Number(footerHeight)) ? footerHeight : `${footerHeight}px`,
-      "--left-railbar-width": isNaN(Number(leftRailbarWidth))
-        ? leftRailbarWidth
-        : `${leftRailbarWidth}px`,
-      "--right-railbar-width": isNaN(Number(rightRailbarWidth))
-        ? rightRailbarWidth
-        : `${rightRailbarWidth}px`,
-      "--left-drawer-width": isNaN(Number(leftDrawerWidth))
-        ? leftDrawerWidth
-        : `${leftDrawerWidth}px`,
-      "--right-drawer-width": isNaN(Number(rightDrawerWidth))
-        ? rightDrawerWidth
-        : `${rightDrawerWidth}px`,
-    },
-    userStyles
-  );
+  const isNumeric = (measure: string | number) =>
+    typeof measure === "number" || !isNaN(Number(measure));
 
-  $: classes = createClasses(["q-layout", userClasses]);
+  const getStyleValue = (measure: string | number) =>
+    isNumeric(measure) ? `${measure}px` : measure;
```

Comment:
`useSize` does that too

Reply:
Hmm, this is a refactor of the conditions in this component. Not gonna change it now, but we should look at it again in the future. It seems `useSize` also has some nuances (`+size > 0 ?`) that could or could not cause different effects.